packages/backend-function/src/factory.ts

```diff
@@ -260,14 +263,43 @@ class AmplifyFunction
       },
     });
 
-    if (secretPolicyStatement) {
-      functionLambda.grantPrincipal.addToPrincipalPolicy(secretPolicyStatement);
-    }
-
     this.resources = {
       lambda: functionLambda,
     };
+    this.lambda = functionLambda;
+
+    // this is a bit of a hack in order to late-bind resolution of ssm environment variables to the end of the synth process
+    this.node.addValidation({
```

Comment:
This seems to be the simplest way to delay the resolution of these values until after the CDK constructors execute, but before synth runs. See https://docs.aws.amazon.com/cdk/api/v2/docs/constructs.Node.html#addwbrvalidationvalidation

This gives a chance for getResourceAccessAcceptor to add environment variables and permissions before all of this config is finalized

Reply:
Good call, let me see if I can use that.